Packages/com.unity.multiplayer.samples.coop/Utilities/Testing/TestUtilities.cs

```diff
@@ -0,0 +1,163 @@
+        {
+            var buttonGameObject = GameObject.Find(name);
+
+            Assert.IsNotNull(buttonGameObject,
+                $"Button GameObject with name {name} not found in scene!");
+
+            EventSystem.current.SetSelectedGameObject(buttonGameObject);
+
+            var buttonComponent = buttonGameObject.GetComponent<Button>();
+
+            Assert.IsNotNull(buttonComponent, $"Button component not found on {buttonGameObject.name}!");
+
+            buttonComponent.onClick.Invoke();
+        }
+
+        /// <summary>
+        /// Helper wrapper method for asserting the completion of a scene load to be used inside Playmode tests. A scene
+        /// is either loaded successfully, or the loading process has timed out.
+        /// </summary>
+        /// <param name="sceneName"> Name of scene </param>
+        /// <returns> IEnumerator to track scene load process </returns>
+        public static IEnumerator AssertIsSceneLoaded(string sceneName)
+        {
+            var waitUntilSceneLoaded = new WaitForSceneLoad(sceneName);
+
+            yield return waitUntilSceneLoaded;
+
+            Assert.That(!waitUntilSceneLoaded.TimedOut);
+        }
+
+        /// <summary>
+        /// Helper wrapper method for asserting the completion of a scene load to be used inside Playmode tests. A scene
+        /// is either loaded successfully, or the loading process has timed out.
+        /// </summary>
+        /// <param name="sceneName"> Name of scene </param>
+        /// <param name="networkSceneManager"> NetworkSceneManager instance </param>
+        /// <returns> IEnumerator to track scene load process </returns>
+        public static IEnumerator AssertIsNetworkSceneLoaded(string sceneName, NetworkSceneManager networkSceneManager)
+        {
+            Assert.That(networkSceneManager != null, "NetworkSceneManager instance is null!");
+
+            var waitForNetworkSceneLoad = new WaitForNetworkSceneLoad(sceneName, networkSceneManager);
+
+            yield return waitForNetworkSceneLoad;
+
+            Assert.That(!waitForNetworkSceneLoad.TimedOut);
+        }
+
+        /// <summary>
+        /// Custom IEnumerator class to validate the loading of a Scene by name. If a scene load lasts longer than
+        /// k_MaxSceneLoadDuration it is considered a timeout.
+        /// </summary>
+        class WaitForSceneLoad : CustomYieldInstruction
+        {
+            string m_SceneName;
+
+            float m_LoadSceneStart;
+
+            float m_MaxLoadDuration;
+
+            public bool TimedOut { get; private set; }
+
+            public override bool keepWaiting
+            {
+                get
+                {
+                    var scene = SceneManager.GetSceneByName(m_SceneName);
+
+                    var isSceneLoaded = scene.IsValid() && scene.isLoaded;
+
+                    if (Time.time - m_LoadSceneStart >= m_MaxLoadDuration)
+                    {
+                        TimedOut = true;
```

Comment:
in both your custom yields, wouldn't you want to throw or at least make your test fail if there's a timeout?

Reply:
Future test writing is not going to be done only by you, so making sure this API is clear straightforward is important.
What's the most likely scenario? Waiting for scene load and failing if it times out? Or waiting for scene load and catching the exception in case it times out to make still pass?
I'd design this API according to the most likely case.

Reply:
after discussion: always throw

Reply:
is the TimedOut bool still needed?

Reply:
Was not necessary due to the exceptions being thrown. Removed.